monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -4249,10 +4249,18 @@ object Task extends TaskInstancesLevel1 {
     *        case you want the [[monix.execution.misc.Local Local]]
     *        variables to be propagated on async boundaries.
     *        Defaults to `false`.
+    * 
+    * @param isolateLocalContextOnRunToFuture should be set to
+    *        `true` if when using `localContextPropagation` you
+    *        want the [[monix.execution.misc.Local Local]] to be
+    *        isolated when running the task otherwise the
+    *        [[monix.execution.misc.Local Local]] will be visible after
+    *        running to a Future. Defaults to `true`.
     */
   final case class Options(
     autoCancelableRunLoops: Boolean,
-    localContextPropagation: Boolean
+    localContextPropagation: Boolean,
+    isolateLocalContextOnRun: Boolean
```

Comment:
A little bikeshedding: 
Should we call it `localContextIsolateOnRun` to be prefix everything with `localContext`? In other PR, Oleg suggested `isolateLocalContextOnFork` for auto-isolation during forking fibers (so `start`, `parMapN` etc.). It's not merged though

Reply:
Sure I can change it to `localContextIsolateOnRun`, it also sounds better.